onnxruntime/core/providers/webnn/builders/model.cc

```diff
@@ -36,8 +36,20 @@ Status Model::Predict(const InlinedHashMap<std::string, OnnxTensorData>& inputs,
     }
     auto num_elements = SafeInt<size_t>(Product(tensor.tensor_info.shape));
     emscripten::val view{emscripten::typed_memory_view(num_elements, static_cast<const float*>(tensor.buffer))};
-    wnn_inputs_.set(name, view);
+    // WebNN API only accepts non-shared ArrayBufferView.
+    // https://webmachinelearning.github.io/webnn/#typedefdef-mlnamedarraybufferviews
+    emscripten::val SharedArrayBuffer = emscripten::val::global("SharedArrayBuffer");
+    if (SharedArrayBuffer.as<bool>()) {
```

Comment:
We should check whether ONNXRuntime is multi-threads build. There is a case that `SharedArrayBuffer` is available, but ONNXRuntime is a single thread build.

Probably using `#if PTHREADS`

Please refer to WebGPU does: https://github.com/emscripten-core/emscripten/blob/384514e2797d1e57f8e5407989ec9ad7a43878d5/src/library_webgpu.js#L1443

Reply:
`PTHREADS` is not defined as macro in onnxruntime. We try to add the `ENABLE_WEBASSEMBLY_THREADS` flag as a macro.